mne/io/cnt/cnt.py

```diff
@@ -0,0 +1,404 @@
+        if r > 1:  # If a point is outside the sphere set z=0.
+            coords.append([x, y, 0])
+            continue
+        alpha = np.arccos(r)
+        z = np.sin(alpha)
+        coords.append([x, y, z])
+    return coords
+
+
+class RawCNT(_BaseRaw):
+    """Raw object from Neuroscan CNT file.
+
+    Note: Channels that are not assigned with keywords ``eog``, ``ecg``,
+    ``emg`` and ``misc`` are assigned as eeg channels. All the eeg channels are
+    fit to a sphere when computing the z-coordinates for the channels.
+    If channels assigned as eeg channels were placed away from the head (i.e.
+    x and y coordinates don't fit to a sphere), all the channel locations will
+    be distorted.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to the CNT file.
+    eog : list | tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If 'auto', the channel names beginning with
+        ``EOG`` are used. Defaults to empty tuple.
+    ecg : list or tuple
+        Names of channels or list of indices that should be designated
+        ECG channels. If 'auto', the channel names beginning with
+        ``ECG`` are used. Defaults to empty tuple.
+    emg : list or tuple
+        Names of channels or list of indices that should be designated
+        EMG channels. If 'auto', the channel names beginning with
+        ``EMG`` are used. Defaults to empty tuple.
+    misc : list or tuple
+        Names of channels or list of indices that should be designated
+        MISC channels. Defaults to empty tuple.
+    read_blocks : bool
+        Whether to read data in blocks. This is for dealing with different
+        kinds of CNT data formats.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    def __init__(self, input_fname, eog=(), ecg=(), emg=(), misc=(),
+                 read_blocks=True, preload=False, verbose=None):
+        input_fname = path.abspath(input_fname)
+        info, cnt_info = _get_cnt_info(input_fname, eog, ecg, emg, misc,
+                                       read_blocks)
+        last_samps = [cnt_info['n_samples'] - 1]
+        super(RawCNT, self).__init__(
+            info, preload, filenames=[input_fname], raw_extras=[cnt_info],
+            last_samps=last_samps, orig_format='int',
+            verbose=verbose)
+
+    @verbose
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
```

Comment:
is it too different from the other `_read_segment_file` functions to consolidate with them?